subsys/dfu_data_writer/src/dfu_data_writer.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdint.h>
+#include <zephyr.h>
+#include <flash.h>
+#include <dfu/mcuboot.h>
+#include <pm_config.h>
+#include <logging/log.h>
+#include <dfu_data_writer.h>
+
+LOG_MODULE_REGISTER(dfu_data_writer, CONFIG_DFU_DATA_WRITER_LOG_LEVEL);
+
+static u32_t	flash_address;
+static struct	device *flash_dev;
+static bool	is_flash_page_erased[FLASH_PAGE_MAX_CNT];
+
+static int flash_page_erase_if_needed(u32_t address)
+{
+	struct flash_pages_info info;
+	int err = flash_get_page_info_by_offs(flash_dev, address, &info);
+
+	if (err != 0) {
+		LOG_ERR("flash_get_page_info_by_offs returned error %d\n", err);
```

Comment:
I suggest making a macro for printing the error from a function, since it's done so many times.

Reply:
I would rather avoid library-specific macros for printing errors.
Nothing wrong with a bit of copy-pasta, it actually shows a good pattern.

Reply:
But it's a bit inconsistent in places :). I'd prefer the macro, but it's a matter of taste I guess, I'll leave it up to @hakonfam to do what he pleases.

Reply:
I fixed up the inconsistencies.